source/config.cpp

```diff
@@ -39,22 +61,64 @@ void Config::loadConfiguration()
     }
 }
 
+/**
+ * @brief Stores current configuration in the config .json file
+ *
+ * @return Nothing
+ */
 void Config::storeConfiguration()
 {
-    QString jsonFilePath = QCoreApplication::applicationDirPath() + "/D1GraphicsTool.config.json";
-
     QFile saveJson(jsonFilePath);
     saveJson.open(QIODevice::WriteOnly);
     QJsonDocument saveDoc(theConfig);
     saveJson.write(saveDoc.toJson());
     saveJson.close();
 }
 
+/**
+ * @brief Creates directories on certain path
+ *
+ * This function creates directories on certain path. Path location depends on
+ * if user is working on Windows or Mac/Linux.
+ *
+ * If the user works on Windows, it will save it under AppData/.config/[...].
+ * On any other OS it will save it under /home/user/.config/diasurgical/[...]
+ *
+ * @return Returns true if path has been created or already existed - false otherwise
+ */
+bool Config::createDirectoriesOnPath()
+{
+#ifdef Q_OS_WINDOWS
+    jsonFilePath = QStandardPaths::AppLocalDataLocation	+ "/.config/diasurgical/d1-graphics-tool/";
+#else
+    jsonFilePath = QDir::homePath() + "/.config/diasurgical/d1-graphics-tool/";
+#endif
```

Comment:
Looks like this generates the propper full path for settings on all platforms so no need for the else or hardcoded values.
```suggestion
    jsonFilePath = QStandardPaths::AppLocalDataLocation;
```
https://doc.qt.io/qt-6/qstandardpaths.html